File: src/exemplos/05_EstruturasDeDados/05_Registro/00-data.py
#  -*- coding: utf-8 -*-
##   @package: 00-data.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo de uso de registros. Em linguagens orientadas a objetos, um registro
# é um tipo especial de "classe" - um dos principais conceitos do paradigma de
# Orientação a Objetos.
# (http://pt.wikipedia.org/wiki/Orienta%C3%A7%C3%A3o_a_objetos)


# Definição do registro:
class Data():
    # Função de inicialização de valores, é executada sempre que uma nova
    # instância é criada.
    def __init__(self):
        dia, mes, ano = 0, 0, 0  # Armazenam a "data" em si.
        descricao = ''           # Armazenam uma descrição da data.


data = Data()  # Como um registro não é um tipo primitivo, é preciso declarar a
               # variável como tal. A instrução "Data()" cria uma nova
               # instância.


data.descricao = input('Digite a descrição: ')  # Armazene a informação lida da
                                                # entrada padrão no espaço
                                                # identificado como "descricao"
                                                # dentro da variável "data"
                                                # (registro).

data.ano = int(input('Digite o ano: '))  # Armazene a informação lida da
                                         # entrada padrão no espaço
                                         # identificado como "ano" dentro da
                                         # variável "data" (registro).

data.mes = int(input('Digite o mês: '))  # Armazene a informação lida da
                                         # entrada padrão no espaço
                                         # identificado como "mes" dentro da
                                         # variável "data" (registro).


data.dia = int(input('Digite o dia: '))  # Armazene a informação lida da
                                         # entrada padrão no espaço
                                         # identificado como "dia" dentro da
                                         # variável "data" (registro).


# Acesse as informações armazenadas e mostre-as.
print('%s:\n%02d/%02d/%04d\n' % (data.descricao,
                                 data.dia,
                                 data.mes,
                                 data.ano))

File: src/exemplos/05_EstruturasDeDados/05_Registro/00-data.c
/**      @file: 00-data.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo de uso de registros. */

#include <stdio.h>

int main() {
  /* Definição da estrutura do registro (identificado como "data"): */
  struct {
    int dia, mes, ano;  /* A "data" em si. */
    char descricao[50]; /* Uma descrição da data. */
  } data;

  /* Armazene a informação lida da entrada padrão no espaço identificado como
    "descricao" dentro da variável "data" (registro). */
  printf("Digite a descrição: ");
  scanf("%[^\n]", data.descricao);

  /* Armazene a informação lida da entrada padrão no espaço identificado como
    "ano" dentro da variável "data" (registro). */
  printf("Digite o ano: ");
  scanf("%d", &(data.ano));

  /* Armazene a informação lida da entrada padrão no espaço identificado como
    "mes" dentro da variável "data" (registro). */
  printf("Digite o mês: ");
  scanf("%d", &(data.mes));

  /* Armazene a informação lida da entrada padrão no espaço identificado como
    "dia" dentro da variável "data" (registro). */
  printf("Digite o dia: ");
  scanf("%d", &(data.dia));

  /* Acesse as informações armazenadas e mostre-as. */
  printf("%s:\n%02d/%02d/%04d\n", data.descricao,
                                  data.dia,
                                  data.mes,
                                  data.ano);

  return 0;
}